### ISSUE TITLE
Refactor: Make `jump_target` a property of SCFG

### ISSUE THREAD

**Issue by kc611** [numba/numba-rvsdg#73]
The attribute `jump_target` of `BasicBlock` class is mutated over the course of various transformations, this leads to the blocks being rebuilt and the SCFG being deep copied every time a transformation is applied.

It should be made a mutable property of the SCFG class instead of an immutable property of the `BasicBlock` frozen data class. This will prevent unnecessary copies.

**Comment by guilhermeleobas** [comment]
Hi @kc611, is this somehow related to #58?

**Comment by kc611** [comment]
Yes, it is what #58 was made to resolve. I'll link the issue shortly.
